### PR TITLE
kie-issues#1506: Use per-package local Maven repositories for dependency declaration correctness on `kie-tools`

### LINKED ISSUE #1506: Use per-package local Maven repositories for dependency declaration correctness on `kie-tools`
To make sure packages can only consume what they declare as a dependency, leveraging [`maven.repo.local.tail`](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=235837219) is essential. 

When building (I.e., `mvn clean verify`), Maven-based packages on `kie-tools` won't install their resulting JARs to `maven.repo.local` anymore, but instead to `dist/1st-party-m2`. This way, the "shared" (a.k.a head) local repository won't contain the artifacts produced by Maven-based packages, making it impossible to access modules that have not been declared as a dependency.

### PR DESCRIPTION
**Pull request by tiagobento**
Closes https://github.com/apache/incubator-kie-issues/issues/1506

---

This PR introduces a new way of working with Maven on `kie-tools`. It takes advantage of the `maven.repo.local.tail` configuration option to allow not installing all packages to `maven.repo.local` and risk depending on undeclared packages.

~This new structure depends on https://github.com/quarkusio/quarkus/pull/43352, so it can only be merged once `kie-tools` is using a version of Quarkus that contains this PR.~

~There are a lot of things to be written about this topic, and I'll get to it once we're closer to being able to merge this.~

I made the ""per-package dist"" way of wiring Maven-based packages together be disabled for now, so we can proceed merging this PR and I'll make it the default once https://github.com/quarkusio/quarkus/pull/43352 is available on Quarkus.

In this PR:
- Created `_intellij-project` to hold files that allow importing `kie-tools` on IntelliJ as a single project.
- Renamed some variables on Webpack and other configuration files standardizing the use of `envs`. We should reserve the name `buildEnv` for the tool not the values.
- Merged `maven-config-setup-helper` into `maven-base`, Making `maven-base` the only requirement for Maven-based packages.
- Configured `-ntp` where possible to save space on logs.
- Stopped copying unnecesasry artifacts inside SonataFlow devmode and builder images.
- Improved <name> and <description> of top-level POMs on Maven-based packages.
- Centralized almost all 3rd party dependencies into maven-base's <dependencyManagement>
- Create and configured `include-1st-party-dependencies` to allow for running `mvn` during bootstrap, when dependencies might not have been built yet.
- Removed a lot of unnecessary configuration on Maven-based packages POMs, which is not centralized on maven-base's POM. (E.g., maven plugins versions)
- Moved `mvnw` installation to `maven-base`, instead of it being a script on package.json. Although I think we could get rid of this entirely.
- Proper cleanup of `m2-repo-via-http-image` on `dev-deployment-kogito-quarkus-blank-app-image/package.json`
- Created `m2-repo-via-http:container:prepare-m2-repo-volume` scripts on packages depending on `maven-m2-repo-via-http-image` to support multiple local Maven repositories, using hard links.
- Removed unnecessary `DEV_DEPLOYMENT_KOGITO_QUARKUS_BLANK_APP_IMAGE__mavenM2RepoViaHttpImage` env var.
- Configured `kn-plugin-workflow` tests to properly wire local test projects with `-Dmaven.repo.local.tail`.
- Renamed `$(build-env quarkusPlatform.version)` to `$(build-env versions.quarkus)`
- Renamed `$(build-env kogitoRuntime.version)` to `$(build-env versions.kogito)`
- Introduced a basic MANUAL.md file containing a little bit of information about the `kie-tools` monorepo.

Comments and questions are welcome! :)

---

~P.S.: It's expected that a few tests will break on `kn-plugin-workflow` because we're using Quarkus 3.8.6, which doesn't contain the PR I mentioned above.~

### CODE REVIEW

**Review by porcelli — APPROVED** [review]

File: packages/maven-base/pom.xml [hunk 1]
```diff
@@ -227,6 +412,103 @@
...
+              </configuration>
+            </plugin>
+          </plugins>
+        </pluginManagement>
+        <plugins>
+          <!--  -->
+          <!-- Removes the local Maven repository from the `dist` folder. -->
+          <!--  -->
+          <plugin>
+            <artifactId>maven-clean-plugin</artifactId>
```
Comment (pefernan) [inline packages/maven-base/pom.xml:451]: don't we need groupId here?

File: packages/maven-base/kie-tools-maven-base.iml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pefernan) [inline packages/maven-base/kie-tools-maven-base.iml:1]: I think this shouldn't be added right?

Comment (tiagobento) [reply packages/maven-base/kie-tools-maven-base.iml:1]: It should. `*.iml` files are what allow us to have pre-configured default for modules of `_intellij-project`.

Comment (pefernan) [reply packages/maven-base/kie-tools-maven-base.iml:1]: then +1

**Review by pefernan — APPROVED** [review]

**Review by thiagoelg — APPROVED** [review]
Awesome job! 🎉 
This solves most of the issues we had when building Java packages on kie-tools. Thanks @tiagobento!

File: _intellij-project/.idea/.gitignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Default ignored files
```
Comment (yesamer) [inline _intellij-project/.idea/.gitignore:1]: @tiagobento Missing License header

File: _intellij-project/.idea/eclipseCodeFormatter.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yesamer) [inline _intellij-project/.idea/eclipseCodeFormatter.xml:1]: @tiagobento Missing License header

File: _intellij-project/.idea/modules.xml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (yesamer) [inline _intellij-project/.idea/modules.xml:2]: @tiagobento Missing License header

File: _intellij-project/.idea/vcs.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<project version="4">
```
Comment (yesamer) [inline _intellij-project/.idea/vcs.xml:2]: @tiagobento Missing License header

File: _intellij-project/maven-based-packages-@-kie-tools.iml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (yesamer) [inline _intellij-project/maven-based-packages-@-kie-tools.iml:1]: @tiagobento Missing License header

File: examples/drools-process-usertasks-quarkus-example/drools-process-usertasks-quarkus-example.iml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yesamer) [inline examples/drools-process-usertasks-quarkus-example/drools-process-usertasks-quarkus-example.iml:1]: @tiagobento Missing License header

File: examples/jbpm-compact-architecture-example/jbpm-compact-architecture-quarkus-example.iml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yesamer) [inline examples/jbpm-compact-architecture-example/jbpm-compact-architecture-quarkus-example.iml:1]: @tiagobento Missing License header

File: examples/sonataflow-greeting-quarkus-example/sonataflow-greeting-quarkus-example.iml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline examples/sonataflow-greeting-quarkus-example/sonataflow-greeting-quarkus-example.iml:2]: @tiagobento Missing License header

File: packages/dashbuilder/dashbuilder-parent.iml [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/dashbuilder/dashbuilder-parent.iml:2]: @tiagobento Missing License header

File: packages/dev-deployment-kogito-quarkus-blank-app/dev-deployment-kogito-quarkus-blank-app.iml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/dev-deployment-kogito-quarkus-blank-app/dev-deployment-kogito-quarkus-blank-app.iml:2]: @tiagobento Missing License header

File: packages/dmn-marshaller-backend-compatibility-tester/dmn-marshaller-backend-compatibility-tester.iml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<module type="JAVA_MODULE" version="4">
```
Comment (yesamer) [inline packages/dmn-marshaller-backend-compatibility-tester/dmn-marshaller-backend-compatibility-tester.iml:2]: @tiagobento Missing License header

File: packages/dmn-testing-models/dmn-testing-models.iml [hunk 14]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/dmn-testing-models/dmn-testing-models.iml:2]: @tiagobento Missing License header

File: packages/extended-services-java/extended-services-java.iml [hunk 15]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/extended-services-java/extended-services-java.iml:2]: @tiagobento Missing License header

File: packages/jbpm-quarkus-devui/jbpm-quarkus-devui-parent.iml [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/jbpm-quarkus-devui/jbpm-quarkus-devui-parent.iml:2]: @tiagobento Missing License header

File: scripts/scripts.iml [hunk 17]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
```
Comment (yesamer) [inline scripts/scripts.iml:2]: @tiagobento Missing License header

File: repo/repo.iml [hunk 18]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
```
Comment (yesamer) [inline repo/repo.iml:2]: @tiagobento Missing License header

File: packages/yard-validator-worker/yard-validator-worker.iml [hunk 19]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/yard-validator-worker/yard-validator-worker.iml:2]: @tiagobento Missing License header

File: packages/yard-model/yard-model.iml [hunk 20]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/yard-model/yard-model.iml:2]: @tiagobento Missing License header

File: packages/stunner-editors/stunner-editors-parent.iml [hunk 21]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/stunner-editors/stunner-editors-parent.iml:2]: @tiagobento Missing License header

File: packages/serverless-workflow-diagram-editor/serverless-workflow-diagram-editor-parent.iml [hunk 22]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
```
Comment (yesamer) [inline packages/serverless-workflow-diagram-editor/serverless-workflow-diagram-editor-parent.iml:2]: @tiagobento Missing License header

**Review by yesamer — COMMENTED** [review]
So far, just some missing headers.
I guess `*.iml` file shouldn't have the headers, because autogenerated by IntellliJ. In such a case, please ignore them

File: _intellij-project/.idea/jarRepositories.xml [hunk 23]
```diff
@@ -1,11 +1,16 @@
...
   <component name="RemoteRepositoriesConfiguration">
     <remote-repository>
       <option name="id" value="central" />
       <option name="name" value="Central Repository" />
       <option name="url" value="https://repo.maven.apache.org/maven2" />
     </remote-repository>
+    <remote-repository>
+      <option name="id" value="jdt.ls.p2" />
+      <option name="name" value="jdt.ls.p2" />
+      <option name="url" value="https://download.eclipse.org/jdtls/milestones/1.30.1/repository/" />
```
Comment (yesamer) [inline _intellij-project/.idea/jarRepositories.xml:12]: @tiagobento This repo is strictly related to the current jdt.ls version used in our code. Said so, I would add a comment here and in the [vscode-java-code-completion-extension-plugin/pom.xml](https://github.com/apache/incubator-kie-tools/pull/2635/files/c467aed95f172806ed5cdb5f76c6dfb072b2b18f..31cd8ad79ffe3e0467178c53d672911512ae79b1#diff-8927d6151a77f4c36f1feb9ea46dedb8f212613b1227b3929ad16a5a390a9795) to warn the user to update the repo when the version is updated.

File: packages/dev-deployment-kogito-quarkus-blank-app/pom.xml [hunk 24]
```diff
@@ -36,19 +36,35 @@
...
+    <version.maven.failsafe.plugin>3.0.0-M7</version.maven.failsafe.plugin>
+    <version.maven.surefire.plugin>3.5.0</version.maven.surefire.plugin>
+    <version.maven.install.plugin>3.1.3</version.maven.install.plugin>
+    <version.maven.deploy.plugin>3.1.2</version.maven.deploy.plugin>
+    <version.maven.site.plugin>3.12.1</version.maven.site.plugin>
+    <version.maven.dependency.plugin>3.6.1</version.maven.dependency.plugin>
+    <version.maven.jar.plugin>3.4.1</version.maven.jar.plugin>
+    <version.maven.remote.resources.plugin>3.2.0</version.maven.remote.resources.plugin>
+    <version.maven.clean.plugin>3.4.0</version.maven.clean.plugin>
+    <version.codehaus.flatten.plugin>1.6.0</version.codehaus.flatten.plugin>
```
Comment (yesamer) [inline packages/dev-deployment-kogito-quarkus-blank-app/pom.xml:60]: As we discussed in private, as a future improvement I would discuss better centralizing the maven plugin management, declaring them in the maven base pom and inheriting them from the apache pom @tiagobento @pefernan

**Review by yesamer — APPROVED** [review]
Great improvement @tiagobento!

**Pull request merged**